internal/lexer/lexer.go

```diff
@@ -63,7 +64,24 @@ func (l *lexer) next() byte {
 	return b
 }
 
-func (l *lexer) run(input []byte, namespace string) (*gostatsd.Metric, *gostatsd.Event, error) {
+func (l *Lexer) reset() {
```

Comment:
Does this need to be thread safe?

Reply:
Explicitly not - hence why every DatagramParser.Run creates its own.